test/test_general.cpp

```diff
@@ -52,3 +53,65 @@ TEST(General, TestLineParsing) {
   EXPECT_EQ(data[1], "Entry2");
   EXPECT_EQ(data[2], "EscapedEntry1;EscapedEntry2");
 }
+
+TEST(General, Dijkstra) {
+  /*
+            (6)
+        .----------.
+       \/          \/
+  0 <-> 1 --> 2 <-> 3
+    (1)   (2)   (3)
+  */
+  CouplingMap cm = {{0, 1}, {1, 0}, {1, 2}, {2, 3}, {3, 2}, {1, 3}, {3, 1}};
+  Matrix      edgeWeights = {
+      {0, 1, 0, 0}, {1, 0, 2, 6}, {0, 15, 0, 3}, {0, 6, 3, 0}};
```

Comment:
Both of these could be declared `const`.

Reply:
Right, thank you